### ISSUE TITLE
API to generate rules?

### ISSUE THREAD

**Issue by flatsiedatsie** [mozilla-iot/gateway#2097]
I suspect that there's an internal API to generate rules that the gateway uses.

Is this API available to add-ons?

I ask because with voice control it becomes very attractive to create simple rules by saying something like "Every day at 11pm turn off the living room lights". Already the Voco add-on allows you to say turn off the lights at 11pm", so it's just a small step away.

If users could create these basic rules by voice, I think it would be best if these would be reflected in the rules overview.

- Does having access to the API also mean the add-on is allowed to use the rules api?
- If so, if there any documentation on how to use this?
- If not, would it be an idea to allow access to add-ons?

This relates to the planned implementation of add-ons that expand the UI (hurray!).
This relates to the proposal to allow users a more ["one-click" WYSIWYG way](https://github.com/mozilla-iot/gateway/issues/2057) of giving add-ons permissions to access devices.

**Comment by mrstegeman** [comment]
I don't believe OAuth currently gives you access to the rules API. @hobinjk is that correct?

I would like to allow extension add-ons to be able to add new rule input/output blocks at some point, but that probably wouldn't require what you're asking for.

**Comment by hobinjk** [comment]
That's correct, but there's nothing preventing it from being implemented. The main thing would be adding more checkboxes to the authorization screen
